File: packages/jsii-dotnet-runtime/test/AWS.Jsii.Tests.Calculator/OverrideReturnsObject.cs
using AWS.Jsii.Runtime.Deputy;

namespace AWS.Jsii.Tests.Calculator
{
    [JsiiClass("jsii-calc", "jsii$jsii_calc$.OverrideReturnsObject", "[]")]
    public class OverrideReturnsObject : DeputyBase
    {
        public OverrideReturnsObject(): base(new DeputyProps(new object[]{}))
        {
        }

        protected OverrideReturnsObject(ByRefValue reference): base(reference)
        {
        }

        protected OverrideReturnsObject(DeputyProps props): base(props)
        {
        }

        [JsiiMethod("test", "{\"primitive\":\"number\"}", "[{\"name\":\"obj\",\"type\":{\"fqn\":\"jsii$jsii_calc$.ReturnsNumber\"}}]")]
        public virtual double Test(IReturnsNumber obj)
        {
            return InvokeMethod<double>(new object[]{obj});
        }
    }
}

File: packages/jsii-pacmak/test/expected.java.jsii-calc/java/org/jsii/tests/calculator/OverrideReturnsObject.java
package org.jsii.tests.calculator;
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.OverrideReturnsObject")
public class OverrideReturnsObject extends org.jsii.JsiiObject {
    protected OverrideReturnsObject(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public OverrideReturnsObject() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    public java.lang.Number test(final org.jsii.tests.calculator.ReturnsNumber obj) {
        return this.jsiiCall("test", java.lang.Number.class, java.util.stream.Stream.of(obj).toArray());
    }
}
